main.py

```diff
@@ -84,19 +66,37 @@ def create_snapshot(machine_name=None):
         if vm_initial_status:
             if session.state == virtualbox.library.SessionState(2):
                 session.unlock_machine()
-    else:
-        print("machine_name is None")
+    except Exception as ex:
+        print(f"Snapshot creation exited prematurely with following exception: {ex}")
+        # This will skip power down of machine on failure
+        return 1
```

Comment:
Since exception is generic, the nature of exception is hard to determine. Therefore, returning 1 to prevent further action to a VM.